### PR TITLE
[Merged by Bors] - Solang integration

### PR DESCRIPTION
**Pull request by kirillt**
This request adds
* Flipper contract, written in Solidity
* JavaScript test-case which uses it
* Solang dependency

Also, this request improves initialization procedure. The idea is that a user needs to have only Cargo, Yarn and Docker/Podman on his/her machine in order to run the testing bench. In case explicit paths are not provided, Solang and Substrate images are pulled and used. WABT is pulled if we can't find it in user's $PATH.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -1,30 +1,75 @@
 #!/bin/bash
 
-set -evu
```
Comment (TriplEight) [inline test.sh:3]: please return this it's essential for CI, `-v` is important to see the output in GitLab, `-u` fails on unset vars and `-e` fails on errors.

**Comment by kirillt** [comment]
bors try

File: test.sh (hunk 1 above)
Comment (kirillt) [reply test.sh:3]: returned

**Comment by kirillt** [comment]
bors try

**Review by pepyakin — APPROVED** [review]

**Pull request closed without merging**